modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java

```diff
@@ -5170,58 +5386,121 @@ public ObjectDefinitionModel newObjectDefinitionModel(long objectFolderId) {
 		objectDefinitionModel.setCreateDate(new Date());
 		objectDefinitionModel.setModifiedDate(new Date());
 
-		// Other field
+		// Other fields
 
-		objectDefinitionModel.setDescriptionObjectFieldId(0);
 		objectDefinitionModel.setObjectFolderId(objectFolderId);
-		objectDefinitionModel.setRootObjectDefinitionId(0);
-		objectDefinitionModel.setTitleObjectFieldId(_counter.get());
-		objectDefinitionModel.setAccountEntryRestricted(false);
 		objectDefinitionModel.setActive(true);
-		objectDefinitionModel.setLabel(
-			StringBundler.concat(
-				"<?xml version=\"1.0\"?><root available-locales=\"en_US\" ",
-				"default-locale=\"en_US\"><Label language-id=\"en_US\">",
-				"Commerce Order</Label></root>"));
-		objectDefinitionModel.setClassName(CommerceOrder.class.getName());
+		objectDefinitionModel.setDBTableName(
+			StringBundler.concat("O_", _companyId, StringPool.UNDERLINE, name));
+		objectDefinitionModel.setLabel(_getLabel(name));
+
+		String className =
+			ObjectDefinitionConstants.
+				CLASS_NAME_PREFIX_CUSTOM_OBJECT_DEFINITION +
+					objectDefinitionModel.getObjectDefinitionId();
+
+		objectDefinitionModel.setClassName(className);
```

Comment:
Does [objectDefinitionModel#getClassName](https://github.com/liferay/liferay-portal/blob/54e81be8b517d4887a4197799d6f690bc840f382/modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectDefinitionModelImpl.java#L977-L984) method use local service?
If I misunderstood something and it is not possible, you can mark this comment as resolved and forward this PR.